backend/internal/flow/engine/flowengine.go

```diff
@@ -169,8 +169,18 @@ func updateContextWithNodeResponse(engineCtx *model.EngineContext, nodeResp *mod
 
 	// Handle authenticated user from the node response
 	if nodeResp.AuthenticatedUser.IsAuthenticated {
+		authnUserAttrs := engineCtx.AuthenticatedUser.Attributes
 		engineCtx.AuthenticatedUser = nodeResp.AuthenticatedUser
 
+		// If engine context already had authenticated user attributes, merge them with the new ones
```

Comment:
Why we need to merge? Isn't the latest one is the most uptodate one?

Reply:
Here actually the attributes from node response is added to the engineCtx. If the same attribute exists in both, nodeCtx one overrides the engine ctx attribute.

Updated the comment with 0a74469574fb98bae85241d073fb528a763728a1

Reply:
One in the engineCtx is probably from the previous step.. So, IMO we can depend on the latest one, rather combining two. But we can revisit this later.

Reply:
We actually depend on the latest one.